src/storage/QueryEdgePropsProcessor.cpp

```diff
@@ -59,14 +51,20 @@ void QueryEdgePropsProcessor::process(const cpp2::EdgePropRequest& req) {
         return;
     }
 
+    int32_t edgeSize = std::accumulate(edgeContexts_.cbegin(), edgeContexts_.cend(), 0,
+                                       [](int ac, auto& ec) { return ac + ec.second.size(); });
+
+    int32_t returnColumnsNum = req.get_return_columns().size() + edgeSize;
     RowSetWriter rsWriter;
     std::for_each(req.get_parts().begin(), req.get_parts().end(), [&](auto& partE) {
         auto partId = partE.first;
         kvstore::ResultCode ret;
         for (auto& edgeKey : partE.second) {
-            ret = this->collectEdgesProps(partId, edgeKey, this->edgeContext_.props_, rsWriter);
-            if (ret != kvstore::ResultCode::SUCCEEDED) {
-                break;
+            for (auto& ec : edgeContexts_) {
```

Comment:
I can't understand this code.  For each edge key,  you could find the related props in edgeContexts_ and collect them.